app/src/composables/use-flows.ts

```diff
@@ -0,0 +1,284 @@
+import api from '@/api';
+import { computed, ref, Ref, provide, inject, ComputedRef } from 'vue';
+import { FlowRaw, Item, PrimaryKey } from '@directus/types';
+import { notify } from '@/utils/notify';
+import { translate } from '@/utils/translate-object-values';
+import { unexpectedError } from '@/utils/unexpected-error';
+import { useCollection } from '@directus/composables';
+import { useFlowsStore } from '@/stores/flows';
+import { useI18n } from 'vue-i18n';
+import { useNotificationsStore } from '@/stores/notifications';
+import formatTitle from '@directus/format-title';
+
+export interface UseFlowsOptions {
+	collection: Ref<string>;
+	primaryKey?: ComputedRef<PrimaryKey | null>;
+	selection?: Ref<Item[]>;
+	location: Ref<'collection' | 'item'>;
+	hasEdits?: Ref<boolean>;
+	onRefreshCallback: () => void;
+}
+
+export interface FlowDialogsContext {
+	confirmButtonCTA: ComputedRef<string>;
+	confirmDialogDetails: ComputedRef<{
+		description: any;
+		fields: any;
+	} | null>;
+	confirmUnsavedChanges: (flowId: string) => void;
+	confirmCustomDialog: (flowId: string) => void;
+	confirmValues: Ref<Record<string, any> | null>;
+	currentFlowId: Ref<string | null>;
+	displayCustomConfirmDialog: ComputedRef<boolean>;
+	displayUnsavedChangesDialog: ComputedRef<boolean>;
+	isConfirmButtonDisabled: ComputedRef<boolean>;
+	resetConfirm: () => void;
+	updateFieldValues: (event: any) => void;
+}
+
+const runManualFlowSymbol = 'runManualFlow';
+
+export function useFlows(options: UseFlowsOptions) {
+	const { collection, hasEdits = ref(false), location, onRefreshCallback, primaryKey, selection = ref([]) } = options;
+
+	const { t } = useI18n();
+	const { primaryKeyField } = useCollection(collection);
+	const flowsStore = useFlowsStore();
+	const notificationStore = useNotificationsStore();
+
+	const runningFlows = ref<string[]>([]);
+	const confirmValues = ref<Record<string, any> | null>(null);
+	const confirmedUnsavedChanges = ref<boolean>(false);
+	const confirmedCustomDialog = ref<boolean>(false);
+	const currentFlowId = ref<string | null>(null);
+
+	const currentFlow = computed(() => {
+		if (!currentFlowId.value) return null;
+
+		return manualFlows.value.find((flow) => flow.id === currentFlowId.value);
+	});
+
+	const currentFlowConfirmations = computed(() => {
+		if (!currentFlow.value) return null;
+
+		return {
+			isUnsavedChangesConfirmationRequired: hasEdits.value,
+			isUnsavedChangesConfirmed: confirmedUnsavedChanges.value,
+			isCustomDialogConfirmationRequired: currentFlow.value.options?.requireConfirmation,
+			isCustomDialogConfirmed: confirmedCustomDialog.value,
+		};
+	});
+
+	const flowDialogsContext = computed(() => ({
+		confirmButtonCTA,
+		confirmDialogDetails,
+		confirmUnsavedChanges,
+		confirmCustomDialog,
+		confirmValues,
+		currentFlowId,
+		displayCustomConfirmDialog,
+		displayUnsavedChangesDialog,
+		isConfirmButtonDisabled,
+		resetConfirm,
+		updateFieldValues,
+	}));
```

Comment:
Let’s pass the value instead of the ref. This prevents us from using `.value` in the template (for example `:model-value="displayUnsavedChangesDialog.value"`).

```suggestion
	const flowDialogsContext = computed(() => ({
		confirmButtonCTA: confirmButtonCTA.value,
		confirmDialogDetails: confirmDialogDetails.value,
		confirmUnsavedChanges,
		confirmCustomDialog,
		confirmValues: confirmValues.value,
		currentFlowId: currentFlowId.value,
		displayCustomConfirmDialog: displayCustomConfirmDialog.value,
		displayUnsavedChangesDialog: displayUnsavedChangesDialog.value,
		isConfirmButtonDisabled: isConfirmButtonDisabled.value,
		resetConfirm,
		updateFieldValues,
	}));
```